src/utils/fetch.ts

```diff
@@ -27,3 +27,15 @@ export const withTimeout = async (
   const res = await Promise.race([asyncCall, timeoutPromise])
   return res
 }
+
+export const fetchUntilSuccess = async (endpoints: string[]): Promise<any> => {
+  // Pick a random endpoint and make a call to that endpoint
+  const endpoint = endpoints[Math.floor(Math.random() * endpoints.length)]
+  console.log('Attempting endpoint: ', endpoint)
```

Comment:
maybe console.info

Reply:
> There's some odd behavior where we're dispatching a ton of requests to fetch metrics. I think it has to do with some broken hook state that we have (if you add some logs in useTopApps, you'll see).
> 
> I don't think this bug is pertinent to this PR, but something we should probably take a look at ASAP.
> 
> Might be the useDiscoveryProviders being called from several components at the same time and that causing state mutation in redux which over-triggers effects.

Yes looked into this and Joe and I found a fix. Can probably tweak the design a bit to avoid this in the future.